FileShareClient/FileShareApiClient.cs

```diff
@@ -106,6 +112,51 @@ public async Task<Stream> DownloadFileAsync(string batchId, string filename)
             }
         }
 
+        public async Task<IResult<DownloadFileResponse>> DownloadFileAsync(string batchId, string fileName, FileStream destinationStream, long fileSizeInBytes = 0, CancellationToken cancellationToken = default)
+        {
+            long startByte = 0;
+            long endByte = fileSizeInBytes < maxDownloadBytes ? fileSizeInBytes - 1 : maxDownloadBytes;
+            var result = new Result<DownloadFileResponse>();
+            HttpStatusCode httpStatusCode = HttpStatusCode.OK;
+
+            while (startByte <= endByte)
+            {
+                string rangeHeader = $"bytes={startByte}-{endByte}";
+
+                var uri = $"batch/{batchId}/files/{fileName}";
+
+                using (var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, uri))
+                {
+                    var httpClient = await GetAuthenticationHeaderSetClient();
+                    if (fileSizeInBytes != 0 && rangeHeader != null)
+                    {
+                        httpRequestMessage.Headers.Add("Range", rangeHeader);
+                        httpStatusCode = HttpStatusCode.PartialContent;
+                    }
+
+                    var response = await httpClient.SendAsync(httpRequestMessage, cancellationToken);
+
+                    await result.ProcessHttpResponse(httpStatusCode, response, true);
+                    if (!result.IsSuccess) return result;
+
+                    using (var contentStream = await response.Content.ReadAsStreamAsync())
+                    {
+                        contentStream.CopyTo(destinationStream);
+                    }
+                }
+                startByte = endByte + 1;
+                endByte += maxDownloadBytes;
+
+                if (endByte > fileSizeInBytes - 1)
+                {
+                    endByte = fileSizeInBytes - 1;
+                }
+
+            }
+            destinationStream.Close();
```

Comment:
Please do not close the stream. The stream is owned by the caller

Reply:
update same, to be handled by caller. Thanks